craystack/rans_test.py

```diff
@@ -48,3 +50,18 @@ def test_flatten_unflatten():
     assert np.all(flat == flat_)
     assert np.all(state[0] == state_[0])
     # assert state[1] == state_[1]
+
+
+def test_base_message():
+    def calc_num_ones(head):
```

Comment:
There should be a NumPy function for this, but there isn't yet. The low level operation is often called 'popcount' or 'population count', see https://github.com/numpy/numpy/issues/16325.